sdk/metric/meter_test.go

```diff
@@ -532,6 +545,106 @@ func TestMeterCreatesInstruments(t *testing.T) {
 	}
 }
 
+func TestMeterWithDropView(t *testing.T) {
+	testCases := []struct {
+		name string
+		fn   func(*testing.T, metric.Meter)
+	}{
+		{
+			name: "SyncInt64Count",
+			fn: func(t *testing.T, m metric.Meter) {
+				ctr, err := m.Int64Counter("sint")
+				assert.NoError(t, err)
+				c, ok := ctr.(x.EnabledInstrument)
+				require.True(t, ok)
+				assert.False(t, c.Enabled(context.Background()))
+			},
+		},
```

Comment:
```go
assert.NoError(t, err)
c, ok := ctr.(x.EnabledInstrument)
require.True(t, ok)
assert.False(t, c.Enabled(context.Background()))
```

is redundant in all tests.

You can do something like

```go
	testCases := []struct {
		name string
		fn   func(*testing.T, metric.Meter) (any, error)
	}{
		{
			name: "SyncInt64Count",
			fn: func(t *testing.T, m metric.Meter) (any, error) {
				return m.Int64Counter("sint")
			},
		},
// ...
	for _, tt := range testCases {
		t.Run(tt.name, func(t *testing.T) {
			rdr := NewManualReader()
			dropView := NewView(
				Instrument{Name: "*"},
				Stream{Aggregation: AggregationDrop{}},
			)
			m := NewMeterProvider(WithReader(rdr), WithView(dropView)).Meter("testInstruments")
			got, err := tt.fn(t, m)
			require.NoError(t, err)
			c, ok := got.(x.EnabledInstrument)
			require.True(t, ok)
			assert.False(t, c.Enabled(context.Background()))
		})
	}
```

Reply:
We can do better. See:

```go
func TestMeterWithDropView(t *testing.T) {
	dropView := NewView(
		Instrument{Name: "*"},
		Stream{Aggregation: AggregationDrop{}},
	)
	m := NewMeterProvider(WithView(dropView)).Meter("testInstruments")

	testCases := []struct {
		name string
		fn   func(*testing.T) (any, error)
	}{
		{
			name: "SyncInt64Count",
			fn: func(*testing.T) (any, error) {
				return m.Int64Counter("sint")
			},
		},
		{
			name: "SyncInt64UpDownCount",
			fn: func(*testing.T) (any, error) {
				return m.Int64UpDownCounter("sint")
			},
		},
		{
			name: "SyncInt64Gauge",
			fn: func(*testing.T) (any, error) {
				return m.Int64Gauge("sint")
			},
		},
		{
			name: "SyncInt64Histogram",
			fn: func(*testing.T) (any, error) {
				return m.Int64Histogram("histogram")
			},
		},
		{
			name: "SyncFloat64Count",
			fn: func(*testing.T) (any, error) {
				return m.Float64Counter("sfloat")
			},
		},
		{
			name: "SyncFloat64UpDownCount",
			fn: func(*testing.T) (any, error) {
				return m.Float64UpDownCounter("sfloat")
			},
		},
		{
			name: "SyncFloat64Gauge",
			fn: func(*testing.T) (any, error) {
				return m.Float64Gauge("sfloat")
			},
		},
		{
			name: "SyncFloat64Histogram",
			fn: func(*testing.T) (any, error) {
				return m.Float64Histogram("histogram")
			},
		},
	}

	for _, tt := range testCases {
		t.Run(tt.name, func(t *testing.T) {
			got, err := tt.fn(t)
			require.NoError(t, err)
			c, ok := got.(x.EnabledInstrument)
			require.True(t, ok)
			assert.False(t, c.Enabled(context.Background()))
		})
	}
}
```

Reply:
applied the refactor